sentry-core/src/main/java/io/sentry/core/ShutdownHookIntegration.java

```diff
@@ -0,0 +1,10 @@
+package io.sentry.core;
+
+/** Registers hook that closes {@link Sentry Sentry SDK} when main thread shuts down. */
+public final class ShutdownHookIntegration implements Integration {
+
+  @Override
+  public void register(IHub hub, SentryOptions options) {
+    Runtime.getRuntime().addShutdownHook(new Thread(Sentry::close));
```

Comment:
H: I'd avoid method references, I already found issues with it depending on the tooling that does desugaring/obfuscation.
Eg: https://github.com/getsentry/sentry-android/blob/master/sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java#L109-L110

Reply:
Also wondering if it's worth to get Runtime as a ctor (package private) param to be able to mock and test it, a ctor overload could pass the `Runtime.getRuntime()` down to the other ctor.

Reply:
👍